client/asset/eth/rpcclient.go

```diff
@@ -8,213 +8,320 @@ package eth
+	"github.com/ethereum/go-ethereum/p2p/enode"
 	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/ethereum/go-ethereum/consensus/misc"
 )
 
-// Check that rpcclient satisfies the ethFetcher interface.
-var _ ethFetcher = (*rpcclient)(nil)
+// Check that nodeClient satisfies the ethFetcher interface.
+var (
+	// https://github.com/ethereum/go-ethereum/blob/16341e05636fd088aa04a27fca6dc5cda5dbab8f/eth/backend.go#L110-L113
+	// ultimately results in a minimum fee rate by the filter applied at
+	// https://github.com/ethereum/go-ethereum/blob/4ebeca19d739a243dc0549bcaf014946cde95c4f/core/tx_pool.go#L626
+	minGasPrice = ethconfig.Defaults.Miner.GasPrice
+
+	_ ethFetcher = (*nodeClient)(nil)
+)
 
-// rpcclient satisfies the ethFetcher interface. Do not use until Connect is
+// nodeClient satisfies the ethFetcher interface. Do not use until Connect is
 // called.
-type rpcclient struct {
-	// c is a direct client for raw calls.
-	c *rpc.Client
-	// ec wraps the client with some useful calls.
-	ec *ethclient.Client
-	n  *node.Node
-	// es is a wrapper for contract calls.
-	es *swap.ETHSwap
+type nodeClient struct {
+	net         dex.Network
+	log         dex.Logger
+	creds       *accountCredentials
+	p2pSrv      *p2p.Server
+	ec          *ethclient.Client
+	node        *node.Node
+	leth        *les.LightEthereum
+	chainID     *big.Int
+	contractors map[uint32]dexeth.Contractor
+	es          *dexeth.ETHSwap
+}
+
+func newNodeClient(dir string, net dex.Network, log dex.Logger) (*nodeClient, error) {
+	node, err := prepareNode(&nodeConfig{
+		net:    net,
+		appDir: dir,
+		logger: log.SubLogger("NODE"),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	leth, err := startNode(node, net)
+	if err != nil {
+		return nil, err
+	}
+
+	creds, err := nodeCredentials(node)
+	if err != nil {
+		return nil, err
+	}
+
+	return &nodeClient{
+		chainID:     big.NewInt(chainIDs[net]),
+		node:        node,
+		leth:        leth,
+		creds:       creds,
+		net:         net,
+		log:         log,
+		contractors: make(map[uint32]dexeth.Contractor),
+	}, nil
+}
+
+func (n *nodeClient) address() common.Address {
+	return n.creds.addr
 }
 
 // connect connects to a node. It then wraps ethclient's client and
 // bundles commands in a form we can easily use.
-func (c *rpcclient) connect(ctx context.Context, node *node.Node, contractAddr *common.Address) error {
-	client, err := node.Attach()
+func (n *nodeClient) connect(ctx context.Context) error {
+	client, err := n.node.Attach()
 	if err != nil {
 		return fmt.Errorf("unable to dial rpc: %v", err)
 	}
-	c.c = client
-	c.ec = ethclient.NewClient(client)
-	c.n = node
-	c.es, err = swap.NewETHSwap(*contractAddr, c.ec)
-	if err != nil {
-		return fmt.Errorf("unable to find swap contract: %v", err)
+	n.ec = ethclient.NewClient(client)
+	n.p2pSrv = n.node.Server()
+	if n.p2pSrv == nil {
+		return fmt.Errorf("no *p2p.Server")
 	}
+
+	n.es, _ = dexeth.NewETHSwap(dexeth.ContractAddresses[0][dex.Simnet], n.ec)
```

Comment:
Relic? (`NewETHSwap` is currently specific to v0 contract API and I think this is all hidden in `Contractor` now, plus Simnet hard coded)